docs/source/format/CDataInterface/PyCapsuleInterface.rst

```diff
@@ -0,0 +1,402 @@
+  support export via Arrow can be recognized by other libraries.
+
+
+Non-goals
+---------
+
+* Standardizing what public APIs should be used for import. This is left up to
+  individual libraries.
+
+
+Comparison to DataFrame Interchange Protocol
+--------------------------------------------
+
+`The DataFrame Interchange Protocol <https://data-apis.org/dataframe-protocol/latest/>`_
+is another protocol in Python that allows for the sharing of data between libraries.
+This protocol is complementary to the DataFrame Interchange Protocol. Many of
+the objects that implement this protocol will also implement the DataFrame
+Interchange Protocol.
+
+This protocol is specific to Arrow-based data structures, while the DataFrame
+Interchange Protocol allows non-Arrow data frames and arrays to be shared as well.
+Because of this, these PyCapsules can support Arrow-specific features such as
+nested columns.
+
+This protocol is also much more minimal than the DataFrame Interchange Protocol.
+It just handles data export, rather than defining accessors for details like
+number of rows or columns.
+
+In summary, if you are implementing this protocol, you should also consider
+implementing the DataFrame Interchange Protocol.
+
+
+Comparison to __arrow_array__ protocol
+--------------------------------------
+
+The ``__arrow_array__`` protocol is a dunder method that defines how PyArrow
+should import an object as an Arrow array. Unlike this protocol, it does
+specific to PyArrow and isn't used by other libraries. It is also limited to
+arrays and does not support schemas, tabular structures, or streams.
+
+PyCapsule Standard
+==================
+
+`PyCapsule`_ allows for a ``name`` to be associated with the capsule, allowing 
+consumers to verify that the capsule contains the expected data. To make sure
+Arrow structs are recognized, the following names should be used:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - C Interface Type
+     - PyCapsule Name
+   * - ArrowSchema
+     - ``arrow_schema``
+   * - ArrowArray
+     - ``arrow_array``
+   * - ArrowArrayStream
+     - ``arrow_array_stream``
+
+
+Lifetime Semantics
+------------------
+
+The exported PyCapsules should have a destructor that calls the
+:ref:`release callback <c-data-interface-released>`
+of the Arrow struct, if it is not already null. This prevents a memory leak in
+case the capsule was never passed to another consumer.
+
+If the capsule has been passed to a consumer, the consumer should have moved
+the data and marked the release callback as null, so there isn’t a risk of
+releasing data the consumer is using.
+:ref:`Read more in the C Data Interface specification <c-data-interface-released>`.
+
+Just like in the C Data Interface, the PyCapsule objects defined here can only
+be consumed once.
+
+
+Export Protocol
+===============
+
+The interface is three separate protocols:
+
+* ``ArrowSchemaExportable``, which defines the ``__arrow_c_schema__`` method.
```

Comment:
Since the protocol classes themselves are optional and do not have any runtime effect, perhaps this should be restructured to first describe the spec itself (the methods and what they do), and then another subsection for optional typing support?

Reply:
For example this is how Python describes the builtin special methods such as `__len__`:
https://docs.python.org/3/reference/datamodel.html#emulating-container-types

Reply:
Thanks. I've refactored to do that. It's quite nice in that I can now reference the methods elsewhere.